src/Compilers/Core/MSBuildTask/GlobalSuppressions.cs

```diff
@@ -0,0 +1,8 @@
+﻿
+// This file is used by Code Analysis to maintain SuppressMessage 
+// attributes that are applied to this project.
+// Project-level suppressions either have no target or are given 
+// a specific target and scoped to a namespace, type, member, etc.
+
+[assembly: System.Diagnostics.CodeAnalysis.SuppressMessage("Performance", "CA1819:Properties should not return arrays", Justification = "MSBuild has lot of properties returning arrays")]
```

Comment:
> Why is this suppressed here instead of with <NoWarn> in the project file

I guess just a preference and ease of application - a suppression code fix offered in IDE only covers adding global suppress message attribute, not a project level no-warn.

> I would be interested in seeing the cases where the warning was reported.

See https://github.com/dotnet/roslyn/blob/master/src/Compilers/Core/MSBuildTask/ManagedCompiler.cs for an example. There are lots of violations in this project.

Reply:
One advantage of `<NoWarn>` is it suppresses even the execution of the rule. I believe the project-level suppression will result in the rule still running followed by getting filtered out later.

Reply:
> I believe the project-level suppression will result in the rule still running followed by getting filtered out later.

AnalyzerDriver only executes an analyzer if `DiagnosticAnalyzer.SupportedDiagnostics` returns at least one supported rule that is effectively enabled in the compilation.

Reply:
I don't necessarily have a preference in how we suppress warnings but I do prefer it be done consintently. I see a lot of NoWarn, manually suppression, assembly attributes and rulesets. Feel like we need to pick a way and go with it. Otherwise it's nearly impossible to determine what rules are in effect because you have to check so many different places to see if they are turned off.